src/main/java/com/zendesk/maxwell/schema/SchemaStore.java

```diff
@@ -354,13 +339,11 @@ private void restoreSchemaMetadata(Connection conn, Long schemaID) throws SQLExc
 			this.base_schema_id = null;
 
 		this.deltas = parseDeltas(schemaRS.getString("deltas"));
-		this.charset = schemaRS.getString("charset");
+		this.schemaVersion = schemaRS.getInt("version");
+		this.schema = new Schema(new ArrayList<Database>(), schemaRS.getString("charset"), this.sensitivity);
```

Comment:
oh you removed the assignment from `restoreFullSchema`, and `restoreDerivedSchema` doesn't seen to use charset, so looks like those changes are fine